src/app/shared/dashboard-list-item/dashboard-list-item.component.spec.ts

```diff
@@ -37,60 +38,374 @@ describe('DashboardListItemComponent', () => {
+    component.isCurrentLoans = true;
+    component.ngOnInit();
+    expect(component.statusText).toEqual('Requested');
+    expect(component.statusIcon).toEqual('calendar');
+    expect(component.statusColor).toEqual('#FFFFFF');
+  });
+
+  it('should be return the correct basic data of a requested loan', () => {
+    component.loan = loanRequested;
+    component.ngOnInit();
+    expect(component.lent).toEqual('0.10 RCN');
+    expect(component.borrowed).toEqual('0.10 RCN');
+    expect(component.repaid).toEqual('-');
+    expect(component.interest).toEqual('-');
+    expect(component.anualRate).toEqual('10%');
+    expect(component.accruedInterest).toEqual('0.00 RCN');
+  });
+
+  it('should be return the correct status of a expired loan', () => {
+    component.loan = loanExpired;
+    component.isCurrentLoans = true;
+    component.ngOnInit();
+    expect(component.statusText).toEqual('Expired');
+    expect(component.statusIcon).toEqual('times');
+    expect(component.statusColor).toEqual('#A3A5A6');
+  });
+
+  it('should be return the correct basic data of a expired loan', () => {
+    component.loan = loanExpired;
+    component.ngOnInit();
+    expect(component.lent).toEqual('2.00 RCN');
+    expect(component.borrowed).toEqual('2.00 RCN');
+    expect(component.repaid).toEqual('-');
+    expect(component.interest).toEqual('-');
+    expect(component.anualRate).toEqual('10%');
+    expect(component.accruedInterest).toEqual('0.00 RCN');
+  });
+
+  it('should be return the correct status of a payed loan', () => {
+    component.loan = loanPayed;
+    component.isCurrentLoans = true;
+    component.ngOnInit();
+    expect(component.statusText).toEqual('Fully Repaid');
+    expect(component.statusIcon).toEqual('check');
+    expect(component.statusColor).toEqual('#59B159');
+  });
+
+  it('should be return the correct basic data of a payed loan', () => {
+    component.loan = loanPayed;
+    component.ngOnInit();
+    expect(component.lent).toEqual('7.00 RCN');
+    expect(component.borrowed).toEqual('7.00 RCN');
+    expect(component.repaid).toEqual('7.10 RCN');
+    expect(component.interest).toEqual('0.1024 RCN');
+    expect(component.anualRate).toEqual('10%');
+    expect(component.accruedInterest).toEqual('0.10 RCN');
+  });
+
   it('should display borrowed label', () => {
-    const expectedValue = component.borrowed;
+    component.loan = loanPayed;
+    component.isBorrowed = true;
+    component.ngOnInit();
     fixture.detectChanges();
+    const label = readComponent(fixture, '#label-borrowed');
+    expect(label.innerText).toBe(component.borrowed);
+  });
 
-    const value = readComponent(fixture, '#label-borrowed');
-    expect(value.innerText).toBe(expectedValue);
+  it('should display repaid label', () => {
+    component.loan = loanPayed;
+    component.isBorrowed = true;
+    component.ngOnInit();
+    fixture.detectChanges();
+    const label = readComponent(fixture, '#label-repaid');
```

Comment:
@martinbobbio use classname instead of IDs

Reply:
Done